core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -310,15 +306,17 @@ private[spark] object Utils extends Logging {
     while (dir == null) {
       attempts += 1
       if (attempts > maxAttempts) {
-        throw new IOException("Failed to create a temp directory (under " + root + ") after " +
-          maxAttempts + " attempts!")
+        throw new IOException(s"Failed to create a temp directory (under $root) after " +
+           s"$maxAttempts attempts!")
       }
       try {
         dir = new File(root, namePrefix + "-" + UUID.randomUUID.toString)
-        if (dir.exists() || !dir.mkdirs()) {
-          dir = null
-        }
-      } catch { case e: SecurityException => dir = null; }
+        // SPARK-35907 Instead of File#mkdirs, Files#createDirectories is expected.
+        Files.createDirectories(dir.toPath)
+      } catch { case e: Exception =>
```

Comment:
nit. This could cause a regression because catching `Exception` can hide lots of things. Is there a way to narrow down this?

Reply:
> nit. This could cause a regression because catching `Exception` can hide lots of things. Is there a way to narrow down this?

Ignoring the exception directly will cause users to be confused about the creation failure, and Files#createDirectories will throw more meaningful exception information, but SecurityException  is not enough to meet the needs.I will take a more appropriate approach.

Reply:
I meant to use a more specific Exception instead of `Ignoring the exception`.

Reply:
This is a good idea @dongjoon-hyun - `Files.createDirectories` lists the exceptions which can be thrown, we can restrict to the subset which need to be caught if they are expected to be ignored.

Reply:
It looks like this conversation was resolved but the issue still persists. I agree with @mridulm, we should just catch `IOException` and `SecurityException` here.

Reply:
I intended to write the code like this:
`case e @ (_ : FileAlreadyExistsException | _ : IOException | _ : SecurityException) =>`
 but considering that the above method: createDirectory(File) also uses Exception, it would be better to keep the same here.And the code is not very concise.

Reply:
I would argue that `createDirectory(File)` above is the one doing it wrong and we should correct it there rather than copying it, but will refer to @dongjoon-hyun and @mridulm here.

Btw, `FileAlreadyExistsException` is an `IOException`, so you can just do
```
case e @ (_ : IOException | _ : SecurityException) =>
```

Reply:
> I would argue that `createDirectory(File)` above is the one doing it wrong and we should correct it there rather than copying it, but will refer to @dongjoon-hyun and @mridulm here.
> 
> Btw, `FileAlreadyExistsException` is an `IOException`, so you can just do
> 
> ```
> case e @ (_ : IOException | _ : SecurityException) =>
> ```

I have a little bit of pursuing code simplicity, thanks to your suggestions, the code now looks much more comfortable.o(^▽^)o